gateway/ln-gateway/src/lib.rs

```diff
@@ -414,6 +418,42 @@ impl Gateway {
         Ok((route_hints, node_pub_key, alias))
     }
 
+    async fn register_clients_timer(&mut self) {
+        let clients = self.clients.clone();
+        let api = self.api.clone();
+        let lnrpc = self.lnrpc.clone();
+        self.task_group
+            .spawn("register clients", move |handle| async move {
+                while !handle.is_shutting_down() {
+                    match Self::fetch_lightning_route_info(lnrpc.clone()).await {
+                        Ok((route_hints, _, _)) => {
+                            for (federation_id, client) in clients.read().await.iter() {
+                                if client
+                                    .register_with_federation(
+                                        api.clone(),
+                                        route_hints.clone(),
+                                        GW_ANNOUNCEMENT_TTL,
+                                    )
+                                    .await
+                                    .is_err()
+                                {
+                                    error!("Error registering federation {federation_id}");
+                                }
+                            }
+                        }
+                        Err(_) => {
+                            error!(
+                                "Could not retrieve route hints, gateway will not be registered."
+                            );
+                        }
+                    }
+
+                    sleep(Duration::from_secs(60)).await;
```

Comment:
I can make this change, I just didn't know if we wanted to support variable length announcement TTLs (i.e gateway for federation1 has a different TTL than for federation2). Since this loop registers the gateway for all federations, the calculation of this delay could get tricky.

Perhaps its simplest to not support than for now and change the delay to be relative to the constant.

Reply:
It's not really dynamic, it's still a constant in the code, but in case that is changed I'd rather have the two be dependent.